src/server/controller/cycleData/getBulkDownload/utils/getMetadata.ts

```diff
@@ -0,0 +1,88 @@
+import { createI18nPromise } from 'i18n/i18nFactory'
+import { i18n as i18nType } from 'i18next'
+import { Objects } from 'utils/objects'
+
+import { Assessment, Cycle, Labels, RowType } from 'meta/assessment'
+
+import { MetadataController } from 'server/controller/metadata'
+
+type Props = {
+  assessment: Assessment
+  cycle: Cycle
+  tableName: string
+  csvColumn: string
+}
+
+/**
+ * Retrieves the unit label path for a given table name and cycle.
+ *
+ * @param {string} tableName - The name of the table.
+ * @param {Cycle} cycle - The cycle.
+ * @returns {string[]} An array representing the path to the unit in the metadata.
+ */
+const getUnitLabelPath = (tableName: string, cycle: Cycle): string[] => {
+  const pathMap: Record<string, string[]> = {
+    growingStockTotal: ['0', 'cols', '0', 'props', 'labels', cycle.uuid],
+    growingStockComposition2025: ['1', 'cols', '0', 'props', 'labels', cycle.uuid],
+    carbonStockSoilDepth: ['0', 'cols', '0', 'props', 'labels', cycle.uuid],
+  }
+
+  // The unit label is found from the second column of the header row by default
+  const defaultPath = ['0', 'cols', '1', 'props', 'labels', cycle.uuid]
```

Comment:
Most of the tables had this format for the unit label.
3 tables had different format